### PR TITLE
[Swift] A bunch of python2/python3 compatibility fixes.

### PR DESCRIPTION
**Pull request by dcci**
Brought to you by yours truly.

<rdar://problem/47806994>

### CODE REVIEW

**Comment by dcci** [comment]
@swift-ci test and merge

File: packages/Python/lldbsuite/test/lang/swift/different_clang_flags/TestSwiftDifferentClangFlags.py [hunk 1]
```diff
@@ -21,10 +20,15 @@
 import os.path
 import unittest2
 
+import sys
+if sys.version_info.major == 2:
+    import commands as subprocess
```
Comment (adrian-prantl) [inline packages/Python/lldbsuite/test/lang/swift/different_clang_flags/TestSwiftDifferentClangFlags.py:25]: Isn't there a subprocess in 2.7, too?

Comment (dcci) [reply packages/Python/lldbsuite/test/lang/swift/different_clang_flags/TestSwiftDifferentClangFlags.py:25]: yes, it just doesn't have the right methods.

**Comment by adrian-prantl** [comment]
Okay.

**Comment by adrian-prantl** [comment]
@swift-ci test and merge

File: packages/Python/lldbsuite/test/lang/swift/different_clang_flags/TestSwiftDifferentClangFlags.py (hunk 1 above)
Comment (adrian-prantl) [reply packages/Python/lldbsuite/test/lang/swift/different_clang_flags/TestSwiftDifferentClangFlags.py:25]: Should we introduce an lldbutil wrapper?

**Comment by dcci** [comment]
@swift-ci test linux

**Review by JDevlieghere — APPROVED** [review]

**Pull request merged**